meta-lmp-base/classes/lmp.bbclass

```diff
@@ -56,3 +56,66 @@ IMAGE_CMD_ostree_append () {
 	# Update default home path
 	sed -i -e 's,:/home,:/var/rootdirs/home,g' usr/etc/passwd
 }
+
+run_fiotool_cmd () {
+	if [ -n "${SOTA_PACKED_CREDENTIALS}" ]; then
+                if [ -e "${SOTA_PACKED_CREDENTIALS}" ]; then
+                        "${1}" -repo "${OSTREE_REPO}" -creds "${SOTA_PACKED_CREDENTIALS}"
+                else
+                        bbwarn "SOTA_PACKED_CREDENTIALS file does not exist."
+                fi
+        else
+                bbwarn "SOTA_PACKED_CREDENTIALS not set. Please add SOTA_PACKED_CREDENTIALS."
+        fi
+
+}
+
+do_image_ostreepush[depends] += "fiotools-native:do_populate_sysroot ca-certificates-native:do_populate_sysroot"
+IMAGE_CMD_ostreepush_lmp () {
+	if [ -n "${USE_FIOTOOLS}" ]; then
+		bbwarn "Pushing repo by using fiopush"
+		run_fiotool_cmd "fiopush"
+	else
+		bbwarn "Pushing repo by using garage-push"
+		# send a copy of the repo manifest to backend if available
+		local SEND_MANIFEST=""
+		# check if garage-push supports the --repo-manifest option before trying
+		if $(garage-push --help | grep -q '^\s*--repo-manifest') && [ -f ${IMAGE_ROOTFS}${sysconfdir}/manifest.xml ]; then
+			SEND_MANIFEST="--repo-manifest ${IMAGE_ROOTFS}${sysconfdir}/manifest.xml"
+		fi
+
+		if [ -n "${SOTA_PACKED_CREDENTIALS}" ]; then
+			if [ -e ${SOTA_PACKED_CREDENTIALS} ]; then
+				garage-push --loglevel 0 --repo=${OSTREE_REPO} \
+					--ref=${OSTREE_BRANCHNAME} \
+					--credentials=${SOTA_PACKED_CREDENTIALS} \
+					--cacert=${STAGING_ETCDIR_NATIVE}/ssl/certs/ca-certificates.crt \
+					$SEND_MANIFEST
+			else
+				bbwarn "SOTA_PACKED_CREDENTIALS file does not exist."
+			fi
+		else
+			bbwarn "SOTA_PACKED_CREDENTIALS not set. Please add SOTA_PACKED_CREDENTIALS."
+		fi
+	fi
+}
+
+do_image_garagecheck[depends] += "fiotools-native:do_populate_sysroot ca-certificates-native:do_populate_sysroot"
+IMAGE_CMD_garagecheck_lmp () {
+	if [ -n "${USE_FIOTOOLS}" ]; then
+		bbwarn "Check if the repo is uploaded by using fiocheck"
+		run_fiotool_cmd "fiocheck"
+	else
+		bbwarn "Check if the repo is uploaded by using garage-check"
+		if [ -n "${SOTA_PACKED_CREDENTIALS}" ]; then
+			# if credentials are issued by a server that doesn't support offline signing, exit silently
+			unzip -p ${SOTA_PACKED_CREDENTIALS} root.json targets.pub targets.sec tufrepo.url 2>&1 >/dev/null || exit 0
+
+			ostree_target_hash=$(cat ${WORKDIR}/ostree_manifest)
+
+			garage-check --ref=${ostree_target_hash} \
+				--credentials=${SOTA_PACKED_CREDENTIALS} \
+				--cacert=${STAGING_ETCDIR_NATIVE}/ssl/certs/ca-certificates.crt
+		fi
+	fi
+}
```

Comment:
We might be able to remove the duplication by prepending and returning if FIOTOOLS, let me do a quick check.

Reply:
ok, I'll check it out.

Reply:
It looks like it's working. This is the run for USE_FIOTOOLS=0 https://ci.foundries.io/projects/msul-dev01/lmp/builds/793/ and this one for USE_FIOTOOLS=1 https://ci.foundries.io/projects/msul-dev01/lmp/builds/794/. It looks fine, the first run pushed to Treehub the second one to OSTreeHub (our solution) in both cases device was updated successfully.

Reply:
Nice.